api/src/modules/business-units/business-unit.repository.ts

```diff
@@ -1,10 +1,61 @@
-import { EntityRepository } from 'typeorm';
+import { EntityRepository, SelectQueryBuilder } from 'typeorm';
 import { BusinessUnit } from 'modules/business-units/business-unit.entity';
 import { ExtendedTreeRepository } from 'utils/tree.repository';
 import { CreateBusinessUnitDto } from 'modules/business-units/dto/create.business-unit.dto';
+import { Logger, NotFoundException } from '@nestjs/common';
+import { SourcingLocation } from 'modules/sourcing-locations/sourcing-location.entity';
 
 @EntityRepository(BusinessUnit)
 export class BusinessUnitRepository extends ExtendedTreeRepository<
   BusinessUnit,
   CreateBusinessUnitDto
-> {}
+> {
+  logger: Logger = new Logger(BusinessUnitRepository.name);
+
+  /**
+   * @description Retrieves business-units and it's ancestors (in a plain format) there are registered sourcingLocations for
+   */
+
+  async getSourcingDataAdminRegionsWithAncestry(): Promise<BusinessUnit[]> {
+    // Join and filters over business-units present in sourcing-locations. Resultant query returns IDs of elements meeting the filters
+    const queryBuilder: SelectQueryBuilder<BusinessUnit> =
+      this.createQueryBuilder('bu')
```

Comment:
Couldn't you just "select distinct sourcingLocation.businessUnitId from sourcingLocation" and avoid the join?

Reply:
Yes I could, the thing is this is split by entities so each repository points to its target entity by default, so I need either add services in Sourcing Location services to retrieve required Ids for different entities, went for this option to keep stuff organised

Or import the standalone manager in each entity with seems nasty as well, at the end of the day I am just retrieving Ids, nothing else from the DB

The trade-off between performance and developer experiences seems good to me. Having said that, let me know if you still want to move this